From aaa66115cc909fcee4a9bd044324ead5d89b01ab Mon Sep 17 00:00:00 2001
From: Sebastian Andrzej Siewior <sebastian@breakpoint.cc>
Date: Fri, 23 Dec 2022 11:36:06 +0100
Subject: Increase FP_MAX_SIZE to 8192.

Increase FP_MAX_SIZE to 8192 in order to handle 4096 bit long RSA keys.
This changes the ABI of the library.

Patch-Name: Increase-FP_MAX_SIZE-to-8192.patch
Signed-off-by: Sebastian Andrzej Siewior <sebastian@breakpoint.cc>
---
 src/headers/tfm.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/headers/tfm.h b/src/headers/tfm.h
index b4bfb8f952788..b1754fb813530 100644
--- a/src/headers/tfm.h
+++ b/src/headers/tfm.h
@@ -99,10 +99,10 @@
 /* Max size of any number in bits.  Basically the largest size you will be multiplying
  * should be half [or smaller] of FP_MAX_SIZE-four_digit
  *
- * You can externally define this or it defaults to 4096-bits [allowing multiplications upto 2048x2048 bits ]
+ * You can externally define this or it defaults to 8192-bits [allowing multiplications upto 4096x4096 bits ]
  */
 #ifndef FP_MAX_SIZE
-   #define FP_MAX_SIZE           (4096+(8*DIGIT_BIT))
+   #define FP_MAX_SIZE           (8192+(8*DIGIT_BIT))
 #endif
 
 /* will this lib work? */
